inf-terraform-aws/Jenkinsfile.template

```diff
@@ -0,0 +1,257 @@
+/* generated jenkins file used for building and deploying AWS-infrastructure in projects */
+
+def final projectId = '@project_id@'
+def final componentId = '@component_id@'
+def final credentialsId = "${projectId}-cd-cd-user-with-password"
+
+def sharedLibraryRepository
+def dockerRegistry
+
+// QS Variables
+
+gitUrl  = <place_your_gitserver_here>    // eg. 'git@git.phoenix.com'
+gitPort = <put_your_gitserver_port_here> // eg.  2222
+
+def preFix
+def stackConfig
+def stackdeploy  = false
+
+def awsAccessKeyId
+def awsSecretAccessKey
+def awsRegion
+def s3Key
+def workspaceKeyPrefix
+
+def awsAccessKeyIdTesting     
+def awsSecretAccessKeyTesting 
+def awsRegionTesting          
+
+def tfvarsjsonfile
+def tfvarsjson
+
+
+@Library('ods-jenkins-shared-library@master') _
+
+odsComponentPipeline(
+  imageStreamTag: 'ods/jenkins-agent-terraform:latest',
+  branchToEnvironmentMapping: [
+    'master': 'dev',
+  ]
+) { context ->
+    stageGetConfig(context)
+    stageInit(context)
+    stageTest(context)
+    stagePlan(context)
+    stageDeploy(context)
+    stageSmoke(context)
+    stageReport(context)
+}
+
+def getEnvSettings(def preFix, def key, def workspace, def region ) {
+  def config = [
+    "GIT_LOGIN_URL=${gitUrl}",
+    "GIT_LOGIN_PORT=${gitPort}",
+    "PREFIX=${preFix}",
+    "S3KEY=${key}"
+  ]
+
+  if (workspace) {
+    config << "TF_WORKSPACE=${workspace}"
+  }
+  if (region) {
+    config << "AWS_DEFAULT_REGION=${region}"
+  }
+  return config
+}
+
+//read and decide stage-environment
+def stageGetConfig(def context) {
+  stage('GetConfig') {
+
+    if (context.environment) {
+      environmentFile         = readYaml file: "./environments/${context.environment}.yml"   
+      environmentFileTesting  = readYaml file: "./environments/testing.yml"
+    
+      stackdeploy  = true
+    
+      awsAccessKeyId      = environmentFile.credentials.key.toLowerCase()
+      awsSecretAccessKey  = environmentFile.credentials.secret.toLowerCase()
```

Comment:
Do we really want to read this from a file?

Reply:
what is your concern?  we only read a reference - the access key itself is stored in OC secrets .

Reply:
Ah cool! I got confused and thought you read the actual secret from the file. All good then.

Maybe I got confused because you read key AND secret. In theory you should also be able to reference just one secret that contains key and secret ... e.g. the basic auth type. What do you think? Might be easier to manage from a user perspective.

Reply:
On that note, I don't find automation / description right now how a user would setup these credentials. Do you have more info on that?

Reply:
good point  ... supposed to be a manual step - helper script creates secrets in OC, user has to adjust values manually